lib/ttfunk/table/os2.rb

```diff
@@ -256,6 +265,80 @@ def group_original_code_points_by_bit(os2)
             end
           end
         end
+
+        def avg_char_width_for(os2, subset)
+          if subset.microsoft_symbol?
+            avg_ms_symbol_char_width_for(os2, subset)
+          else
+            avg_weighted_char_width_for(os2, subset)
+          end
+        end
+
+        def avg_ms_symbol_char_width_for(os2, subset)
+          total_width = 0
+          num_glyphs = 0
+
+          # use new -> old glyph mapping in order to include compound glyphs
+          # in the calculation
+          subset.new2old_glyph.each do |_, old_gid|
+            if (metric = os2.file.horizontal_metrics.for(old_gid))
+              total_width += metric.advance_width
+              num_glyphs += 1 if metric.advance_width > 0
```

Comment:
I wonder why you didn't use that new shiny `Sum` here? :trollface:

Reply:
Hahaha sure, why not!

Reply:
I'm on a fence. On one hand it's good OOP. On the other utility is extremely limited and I used to have this inline, it's familiar inline and utility classes don't make it read much better. My only concern is consistent application. Like here you opted to have sum inline but elsewhere to use the `Sum` class.

Reply:
I think the utility is more obvious in cases where I'm mixing both `Sum` and `Min`/`Max`. I wanted to be able to use `<<` for all of them, since I thought the code was easier to read that way. `Min`/`Max` are definitely the more useful classes, since it's tedious to have to write `val = min < val ? min : val` all over the place. In cases where I'm not using `Min`/`Max`, it doesn't make much sense to use `Sum` by itself. After all, `+=` is built into the Ruby language.